src/app/helpers/MeasurementPlot.tsx

```diff
@@ -35,44 +35,58 @@ export function MeasurementPlot(props: MeasurementPlotProps) {
 function MeasurementComponent(props: MeasurementPlotProps) {
   const {
     measurementDisplay,
-    measurement: { data },
+    measurement,
     dataIndex = 0,
     xVariableName = 'x',
     yVariableName = 'y',
   } = props;
 
-  const { x, y } = useMemo(() => {
-    const { variables } = data[dataIndex];
-    const { [xVariableName]: x, [yVariableName]: y } = variables;
-    if (x === undefined || y === undefined) {
-      throw new Error(
-        `Variable "${
-          x === undefined ? xVariableName : yVariableName
-        }" is not available in data. Only ${Object.keys(
-          data[dataIndex].variables,
-        ).join(', ')} are available`,
-      );
+  const dataXY = useMemo(() => {
+    const dataXY = [];
+    const measurementsArray = Array.isArray(measurement)
+      ? measurement
+      : [measurement];
+    for (const { data } of measurementsArray) {
+      const { variables } = data[dataIndex];
+      const { [xVariableName]: x, [yVariableName]: y } = variables;
+      if (x === undefined || y === undefined) {
+        throw new Error(
+          `Variable "${
+            x === undefined ? xVariableName : yVariableName
+          }" is not available in data. Only ${Object.keys(
+            data[dataIndex].variables,
+          ).join(', ')} are available`,
+        );
+      }
+      dataXY.push({ x, y });
     }
-
-    return { x, y };
-  }, [data, dataIndex, xVariableName, yVariableName]);
-
-  const { color } = measurementDisplay;
-  if (color.kind !== 'fixed') {
-    throw new Error(`unimplemented stroke for kind ${color.kind}`);
-  }
+    return dataXY;
+  }, [dataIndex, measurement, xVariableName, yVariableName]);
 
   return (
     <BasicComponent {...props}>
-      <LineSeries
-        lineStyle={{
-          stroke: color.color,
-        }}
-        data={xyToXYObject({
-          x: x.data,
-          y: y.data,
-        })}
-      />
+      {dataXY.map(({ x, y }, i) => {
+        const { color } = Array.isArray(measurementDisplay)
+          ? measurementDisplay[i]
+          : measurementDisplay;
+        if (color.kind !== 'fixed') {
+          throw new Error(`unimplemented stroke for kind ${color.kind}`);
+        }
+
+        return (
+          <LineSeries
+            // eslint-disable-next-line react/no-array-index-key
+            key={i}
```

Comment:
Measurements have ids, no?

Reply:
Yes true, i will check if we can access to ids here